### PR TITLE
Check for divide by zero in expression eval.

### LINKED ISSUE #11: Crash with divide by zero in literals
SIGFPE at parser.cpp:567 when division by zero occurs in a symbol computation. This should be handled gracefully.

Test case:
```z80
.db 0/0
```

### PR DESCRIPTION
**Pull request by tari**
Fixes #11.

### CODE REVIEW

File: parser.cpp [hunk 1]
```diff
@@ -498,6 +498,7 @@ static const char *parse_num_full (const char *expr, int *value, int depth) {
 	int total = 0, last_num;
 	MATHOP last_op = M_NONE;
 	bool invert_lastnum, neg_lastnum;
+	const char *const expr_start = expr;
```
Comment (adriweb) [inline parser.cpp]: Very minor thing: `const` twice.

Comment (tari) [inline parser.cpp]: That's by design. The value of `expr_start` should never be changed within the function, nor should what it points to.

Comment (adriweb) [inline parser.cpp]: Oh, I see. Thanks.

**Comment by tari** [comment]
While scanning the parser code I noticed it was also vulnerable to crash with `INT_MIN/-1` and might invoke UB due to integer overflow in expression evaluation. I fixed some of it, as outlined in the commit message.

There were some nice buffer overflow vectors in parsing of literal values too, which I got rid of.

**Pull request merged**